client/src/containers/projects/custom-project/store.ts

```diff
@@ -0,0 +1,13 @@
+import { atom } from "jotai";
+import { parseAsStringLiteral, useQueryState } from "nuqs";
+
+import { CASH_FLOW_VIEWS } from "@/containers/projects/custom-project/annual-project-cash-flow/header/tabs";
+
+export const showCostDetailsAtom = atom<boolean>(false);
+
+export function useProjectCashFlowView() {
+  return useQueryState(
+    "cashflow",
+    parseAsStringLiteral(CASH_FLOW_VIEWS).withDefault("chart"),
+  );
+}
```

Comment:
I am aware there's no thumb of rule for this, but I would put the `store` next to the page so it's easier to notice the state of the page is handled there.